crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -638,8 +644,34 @@ fn pending_validator_exchange_rate(
         })
         .collect::<Vec<(IotaAddress, ObjectID, ObjectID, u64, bool)>>();
 
+    // Try to find for validator candidate and get its exchange rates
+    for (_, df) in state.get_dynamic_fields(
+        system_state_summary.validator_candidates_id,
+        None,
+        system_state_summary.validator_candidates_size as usize,
+    )? {
+        let raw_address =
+            String::deserialize(df.name.value).map_err(|err| IotaError::from(err.to_string()))?;
+        let validator_address = IotaAddress::from_str(&raw_address)?;
+
+        let validator_summary = get_validator_from_table(
+            object_store,
+            system_state_summary.validator_candidates_id,
+            &validator_address,
+        )?;
+
+        tables.push((
+            validator_summary.iota_address,
+            validator_summary.staking_pool_id,
+            validator_summary.exchange_rates_id,
+            validator_summary.exchange_rates_size,
+            false,
+        ));
+    }
```

Comment:
I would suggest splitting in a dedicated function `candidate_validator_exchange_rate` to minimize the scope.

Reply:
As suggested the [two functions](https://github.com/iotaledger/iota/pull/4313/commits/9cf72f246047bee46937cce23431ce6d5bf20a52#diff-0a10b5dd25f3cf15826ae9c8e8ab585ac0e76d35d7c0f84a9334f7ed722b5406R206-R209) are separate